torch/optim/lr_scheduler.py

```diff
@@ -965,11 +965,16 @@ class ReduceLROnPlateau(LRScheduler):
             quantity monitored has stopped increasing. Default: 'min'.
         factor (float): Factor by which the learning rate will be
             reduced. new_lr = lr * factor. Default: 0.1.
-        patience (int): Number of epochs with no improvement after
-            which learning rate will be reduced. For example, if
-            `patience = 2`, then we will ignore the first 2 epochs
-            with no improvement, and will only decrease the LR after the
-            3rd epoch if the loss still hasn't improved then.
+        patience (int): The number of allowed epochs with no improvement after
+            which the learning rate will be reduced.
+            For example, with ``patience = 0``:
+            - 1st epoch: A baseline is established. As there's no previous baseline, the first epoch is always considered good.
+            - 2nd epoch: If this epoch performs worse than the baseline, num_bad_epochs increases by 1.
+            Since ``num_bad_epochs = 1 > patience = 0``, the learning rate will be reduced at the end of this epoch
+            and num_bad_epochs resets to 0.
+            - 3rd epoch onwards: If the epoch continues to perform worse than the baseline,
+            num_bad_epochs increases by 1. As ``num_bad_epochs = 1 > patience = 0``, the learning rate is reduced
+            and num_bad_epochs resets to 0 after each epoch.
```

Comment:
```suggestion
            For example, with ``patience = 0``, the case of having no patience:
            - 1st epoch: A baseline is established. As there's no previous baseline, the first epoch is always considered good.
            - 2nd epoch: If this epoch performs worse than the baseline, we now have one untolerated epoch,  and 1 > 0, thus, the learning rate will be reduced at the end of this epoch.
            - 3rd epoch onwards: If the epoch continues to perform worse than the baseline, the learning rate will keep getting reduced at the end of the epoch. Otherwise, the lr will remain the same.
```

Reply:
It's best to avoid including in-code details, since doc readers will not have that context